wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -155,60 +172,50 @@ private static function entryToDisplayXhtml($id, $entry) {
-		$id = self::convertGuidToId($entry->_id);
-
-		$post = new stdClass();
-		$post->post_title = $entry->mainHeadWord[0]->value;
-		$post->post_name = $id;
-		$post->post_content = self::entryToDisplayXhtml($id, $entry);
-		$post->post_status = 'publish';
-		$post->comment_status = 'closed';
-		$post->post_type = 'post';
-		$post->filter = 'raw'; // important, to prevent WP looking up this post in db!		
-	
+		$post->post_content = $displayXhtml;
+		
 		return $post;
 	}
 
-	 public static function entryToReversal($dictionaryId, $lang, $letter, $entry) {	
+	 public static function entryToReversal($entry, $apiParams) {	
 		//<div class=post><div xmlns="http://www.w3.org/1999/xhtml" class="reversalindexentry" id="g009ab666-43dd-4f2f-ba62-7017417f6b23"><span class="reversalform"><span lang="en">aardvark</span></span><span class="sensesrs"><span class="sensecontent"><span class="sensesr" entryguid="gee1142ec-65f5-4e23-8d95-413685a48c23"><span class="headword"><span lang="mos"><a href="https://www.webonary.org/moore/gee1142ec-65f5-4e23-8d95-413685a48c23">tãnturi</a></span></span><span class="scientificname"><span lang="en">orycteropus afer</span></span></span></span></span></div></div>
 		$id = self::convertGuidToId($entry->_id);
-		$reversal_value = '';
-
-		$definitions = $entry->senses->definitionOrGloss;
-		if (!is_array($definitions)) {
-			$definitions = [$definitions];
-		}
 
-		foreach ($definitions as $definition) {
-			$lowerLetter = strtolower($letter);
-			if (($lang == $definition->lang) && ($lowerLetter == strtolower(substr($definition->value, 0, 1)))) {
-				$reversal_value = $definition->value;
-				break;
+		$reversal = new stdClass();
+		if (empty($apiParams['entryType'])) {
+			// Automatically generated reversal based on entry definition
+			$reversal_value = '';
+			$definitions = $entry->senses->definitionOrGloss;
+			if (!is_array($definitions)) {
+				$definitions = [$definitions];
 			}
-		}
 	
-		$content = '<div class="reversalindexentry">';
-		$content .= '<span class="reversalform"><span lang="' . $lang . '">';
-		$content .= $reversal_value . '</span></span>';
+			foreach ($definitions as $definition) {
+				$lowerLetter = strtolower($apiParams['text']);
+				if (($apiParams['lang']== $definition->lang) && ($lowerLetter == strtolower(substr($definition->value, 0, 1)))) {
+					$reversal_value = $definition->value;
+					break;
+				}
+			}
 		
-		$content .= '<span class="sensesrs"><span class="sensecontent">';
-		$content .= '<span class="sensesr" entryguid="' . $id . '">';
-	
-		$content .= '<span class="headword"><span lang="' . $entry->mainHeadWord[0]->lang . '">'
-			. '<a href="' . get_site_url() . '/' . $id . '">' . $entry->mainHeadWord[0]->value . '</a></span></span>';
-	
-		$content .= '</span></span></span>';
-		$content .= '</<div>';
+			$displayXhtml = '<div class="reversalindexentry">';
+			$displayXhtml .= '<span class="reversalform"><span lang="' . $lang . '">';
```

Comment:
`$lang` is undefined.

Reply:
all of this legacy code has been removed